### PR TITLE
Refactor address_codec to render Address serialization consistent with Lotus

### PR DESCRIPTION
**Pull request by ekovalev**
Signed-off-by: Eugene Kovalev <eugene.a.kovalev@gmail.com>

### Description of the Change

Filecoin specification defines an Address as a product type of a network (mainnet or testnet) and a payload. However, existing Lotus implementation does not take the network part of the Address into account (furthermore, it is not even stored in the Address object). Subsequently, the network information is omitted during address serialization into a byte array and only appears in a human-readable string representation.

In order to be consistent with the way the addresses are passed around in Lotus testnet, the network byte is removed from the byte encoding of an address.

Apart from the above CBOR encoding is added for the Address which will be needed for some upcoming changed.

### Benefits

Consistency with Lotus network.

### Possible Drawbacks 

Potential discrepancy with the specification. This needs to be addressed separately.

### CODE REVIEW

File: core/primitives/address/impl/address_builder_impl.cpp [hunk 1]
```diff
@@ -21,14 +21,14 @@ namespace fc::primitives::address {
   outcome::result<Address> AddressBuilderImpl::makeFromSecp256k1PublicKey(
       Network network, const Sec256k1PublicKey &public_key) noexcept {
     OUTCOME_TRY(hash, blake2b_160(public_key));
-    std::vector<uint8_t> sec256k1_bytes{network, Protocol::SECP256K1};
```
Comment (turuslan) [inline core/primitives/address/impl/address_builder_impl.cpp]: Can we avoid calling `decode`?
Something like `return Address{MAINNET, Secp256k1PublicKeyHash(hash)};`
Same for bls.

Also may add TODO for [FIL-118](https://soramitsu.atlassian.net/browse/FIL-118).

File: test/core/primitives/address/address_verifier_test.cpp [hunk 2]
```diff
@@ -183,7 +183,7 @@ TEST_P(AddressVerifierParametrizedTest, GenerateSecp256k1Address) {
   EXPECT_OUTCOME_TRUE(
       res, address_verifier->verifySyntax(address, keypair.public_key));
   ASSERT_TRUE(res);
-  ASSERT_EQ(GetParam(), address.network);
+  ASSERT_EQ(Network::TESTNET, address.network);
```
Comment (Alexey-N-Chernyshov) [inline test/core/primitives/address/address_verifier_test.cpp]: Please, remove parametrized test as it is not used anymore.

Comment (Alexey-N-Chernyshov) [reply test/core/primitives/address/address_verifier_test.cpp]: I mean `INSTANTIATE_TEST_CASE_P()` and `TEST_P` -> `TEST_F`

File: core/primitives/address/impl/address_builder_impl.cpp (hunk 1 above)
Comment (Alexey-N-Chernyshov) [reply core/primitives/address/impl/address_builder_impl.cpp]: Secp256k1 and BLS Addresses are constructed in different ways. Secp256k1 uses hash of public key when bls just copies it. In my opinion code of creation shouldn't duplicate existing code even if it simple.

**Pull request merged**